backend/src/controllers/FriendController.ts

```diff
@@ -0,0 +1,202 @@
+      await friend.save();
+
+      return res.status(200).send({
+        message: 'Friend removed successfully!',
+      });
+    } catch (err) {
+      console.error(err);
+      res.status(500).send({
+        message: `Error while getting User ${userID}`,
+        error: err,
+      });
+    }
+  }
+
+  public static async sendFriendRequest(req: any, res: any) {
+    const userID: string = req.user.data._id;
+    const friendID: string = req.params.friendID;
+    try {
+      const user = await User.findOne({ _id: userID }).exec();
+      const friend = await User.findOne({ _id: friendID }).exec();
+      if (!user || !friend) {
+        return res.status(404).send({
+          message: "One of the users doesn't exist!",
+        });
+      }
+
+      if (!user.friendRequests.includes(friendID)) {
+        user.friendRequests.push(friendID);
+        await user.save();
+      }
+
+      if (!friend.friendRequests.includes(userID)) {
+        friend.friendRequests.push(userID);
+        await friend.save();
+      }
+
+      return res.status(200).send({
+        message: 'Friend request sent successfully!',
+      });
+    } catch (err) {
+      console.error(err);
+      res.status(500).send({
+        message: `Error while getting User ${userID}`,
+        error: err,
+      });
+    }
+  }
+
+  public static async removeFriendRequest(req: any, res: any) {
+    const userID: string = req.user.data._id;
+    const friendID: string = req.params.friendID;
+    try {
+      const user = await User.findOne({ _id: userID }).exec();
+      const friend = await User.findOne({ _id: friendID }).exec();
+      if (!user || !friend) {
+        return res.status(404).send({
+          message: "One of the users doesn't exist!",
+        });
+      }
+
+      if (user.friendRequests.includes(friendID)) {
+        user.friendRequests = user.friendRequests.filter(
+          (id) => id !== friendID
+        );
+        await user.save();
+      }
+
+      if (friend.friendRequests.includes(userID)) {
+        friend.friendRequests = friend.friendRequests.filter(
+          (id) => id !== userID
+        );
+        await friend.save();
+      }
+
+      return res.status(200).send({
+        message: 'Removed friend request successfully!',
+      });
+    } catch (err) {
+      console.error(err);
+      res.status(500).send({
+        message: `Error while getting User ${userID}`,
+        error: err,
+      });
+    }
+  }
+
+  public static async acceptFriendRequest(req: any, res: any) {
+    const userID: string = req.user.data._id;
+    const friendID: string = req.params.friendID;
+    try {
+      const user = await User.findOne({ _id: userID }).exec();
+      const friend = await User.findOne({ _id: friendID }).exec();
+      if (!user || !friend) {
+        return res.status(404).send({
+          message: "One of the users doesn't exist!",
+        });
+      }
+      if (
+        !user.friendRequests.includes(friendID) &&
```

Comment:
I'd suggest making this an `||` instead of a `&&`, so the error is returned in case any of the requests is missing.